flang/include/flang/Lower/ConvertExpr.h

```diff
@@ -99,6 +103,67 @@ createSomeArrayBox(AbstractConverter &converter,
                    const evaluate::Expr<evaluate::SomeType> &expr,
                    SymMap &symMap, StatementContext &stmtCtx);
 
+/// Generate a unique name for a typed literal constant value.
+class LiteralNameHelper {
```

Comment:
Can this be moved to Mangler.cpp? Or maybe to FIRBuilder?

Reply:
FirBuilder looks to me to be the better match.